display offending text going over limit

zkhan122 opened the pull request:
Added print statement to display text going over limit in tts/xtts/tokenizer.py.

ajk1402 committed:
feat(openvoice): initial integration

eginhard committed:
docs(readme): mention openvoice vc

eginhard committed:
feat(openvoice): add config classes

eginhard committed:
refactor(openvoice): remove duplicate and unused code

eginhard committed:
feat(vits): add tau parameter to posterior encoder

eginhard committed:
feat: add openvoice vc model

eginhard committed:
chore: add type hints, fix typos

eginhard committed:
feat(vc): add knnvc model

eginhard committed:
build: switch to forked coqpit

JohnnyStreet committed:
refactor(xtts): update parameters for transformers>=4.47

eginhard committed:
feat(openvoice): add to .models.json

eginhard committed:
ci: validate .models.json file

eginhard committed:
test(freevc): remove unused code

eginhard committed:
refactor(losses): move shared losses into losses.py

eginhard committed:
refactor(wavernn): remove duplicate Stretch2d

I checked that the implementations are the same

eginhard committed:
refactor(freevc): remove duplicate code

eginhard committed:
refactor(xtts): use tortoise conditioning encoder

eginhard committed:
refactor(vocoder): remove duplicate function

eginhard committed:
refactor(tacotron): remove duplicate function

eginhard committed:
refactor: move exists() and default() into generic_utils

eginhard committed:
refactor(xtts): use existing cleaners

eginhard committed:
refactor: move shared function into dataset.py

eginhard committed:
ci: merge integration tests back into unit tests

eginhard committed:
refactor: move more audio processing into torch_transforms

eginhard committed:
test: move test_helpers.py to fast unit tests

eginhard committed:
refactor: move duplicate alignment functions into helpers

eginhard committed:
refactor: remove duplicate to_camel

eginhard committed:
refactor: import get_last_checkpoint from trainer.io

eginhard committed:
fix(api): clearer error message when model doesn't support VC

eginhard committed:
refactor: only use keyword args in Synthesizer

eginhard committed:
test(openvoice): add sanity check

eginhard committed:
test(freevc): fix output length check

eginhard committed:
refactor(vc): rename TTS.vc.modules to TTS.vc.layers for consistency

Same as in TTS.tts and TTS.vocoder

eginhard committed:
Merge pull request #172 from idiap/deduplicate

Remove unused code

eginhard committed:
fix(xtts): clearer error message when file given to checkpoint_dir

eginhard committed:
refactor(bin.synthesize): use Python API for CLI

eginhard committed:
refactor(synthesizer): set sample rate in loading methods

eginhard committed:
chore: bump version to 0.25.0

eginhard committed:
Merge pull request #183 from idiap/openvoice

Add OpenVoice VC models

eginhard committed:
Merge pull request #110 from idiap/uv

build: switch to forked coqpit

eginhard committed:
docs: use .to("cuda") instead of deprecated gpu=True

eginhard committed:
refactor(api): require keyword arguments except for model_name

eginhard committed:
feat(api): support specifying vocoders by name

eginhard committed:
fix(pypi-release): fix publishing workflow (#191)

eginhard committed:
chore(bin.synthesize): remove unused argument

eginhard committed:
feat(api): support passing a custom speaker encoder by path

eginhard committed:
feat(api): allow mixing TTS and vocoder model name and path

eginhard committed:
chore(api): add type hints

eginhard committed:
refactor(xtts): remove duplicate xtts audio config

eginhard committed:
feat(api): support passing speaker/language id file paths

eginhard committed:
Merge pull request #184 from idiap/xtts-error

fix(xtts): clearer error message when file given to checkpoint_dir

eginhard committed:
refactor(api): use save_wav() from Synthesizer instance

eginhard committed:
Merge pull request #197 from idiap/api

Expand Python API capabilities

eginhard committed:
fix: handle difference in xtts/tortoise attention (#199)

eginhard committed:
chore: bump version to 0.25.1 (#202)

eginhard committed:
refactor: move device property into BaseTrainerModel

eginhard committed:
feat(hifigan): add optional linear transformation at the beginning

Used by the WavLM-HiFiGAN vocoder from kNN-VC to change feature dimensions

eginhard committed:
refactor(vc): remove unused argument

eginhard committed:
feat: support vocoders for voice conversion

So far, FreeVC and OpenVoice are both Vits-based and don't have a separate
vocoder. kNN-VC needs to be combined with a Hifigan.

eginhard committed:
feat(synthesizer): show timing for voice conversion

eginhard committed:
build(docs): update dependencies, fix makefile

eginhard committed:
docs: improve documentation

eginhard committed:
docs: move project structure from readme into documentation

eginhard committed:
docs: use nested contents for easier overview

eginhard committed:
docs: streamline readme and reuse content in other docs pages

[ci skip]

eginhard committed:
Merge pull request #207 from idiap/docs

Improve documentation

eginhard committed:
test: switch from nose2 to pytest

eginhard committed:
test(xtts): use temp folder and pytest.parametrize

eginhard committed:
test: use temp output folders, pytest.parametrize, fixtures

eginhard committed:
test(find_unique_phonemes): remove duplicate test

eginhard committed:
test: consistently name test config.json files

eginhard committed:
test(bin.compute_statistics): convert to python

eginhard committed:
test(zoo): use pytest.parametrize to parallelize zoo tests

Also only run zoo tests for the latest Python version

eginhard committed:
feat: allow both Path and strings where possible and add type hints

eginhard committed:
docs: add notes about xtts fine-tuning

eginhard committed:
test: call cli tests via main functions to get test coverage

eginhard committed:
test(zoo): move big model tests into separate files

Makes it easier to run them locally: make test_zoo_big

eginhard committed:
test: move vocoder training tests into one file

Gets rid of all the duplication in the tests

eginhard committed:
Merge pull request #210 from idiap/manager

feat: allow both Path and strings where possible and add type hints

eginhard committed:
feat(manager): print download location when listing models (#213)

eginhard committed:
test: fix wavegrad test and switch to pytest

Need to ensure that gradients are enabled due to the manual training steps here

eginhard committed:
test(bin.extract_spectrograms): run cli test from python

eginhard committed:
test(bin.find_unique_phonemes): run cli test from python

eginhard committed:
docs(xtts): show manual inference with default speakers

eginhard committed:
fix(bin): log to stdout in cli tools, unless pipe_out is set

This way the outputs are available for further downstream processing, e.g. with
grep. For TTS/bin/synthesize.py, if --pipe_out is set, log to stderr because
then only the output audio stream should be on stdout, e.g. to pipe it to aplay.

eginhard committed:
Merge pull request #217 from idiap/stdout

fix(bin): log to stdout in cli tools

eginhard committed:
fix(notebooks): update plot_embeddings_umap notebook

eginhard committed:
refactor(bin.compute_embeddings): factor out arg parsing

eginhard committed:
fix(xtts): voice_dir should remain None if not specified (#224)

eginhard committed:
refactor: use @torch.inference_mode() instead of @torch.no_grad()

SkaceKamen committed:
fix(xtts): use correct language code for Czech num2words call (#237)

* Fix num2words call using non-standard lang code

* build: update minimum num2words version

---------

Co-authored-by: Enno Hermann <enno.hermann@idiap.ch>

eginhard committed:
chore: remove unused callback code (#229)

The callbacks are in trainer.callbacks now

lostways committed:
fix: convert >35 digit English numbers digit-by-digit (#240)

DelovoiDC committed:
docs: update docker image urls (#243)

* Change old docker image url to the one that is relevant to this repo

* Change old docker image url to the one that is relevant to this repo in docker_images.md

* Fixed mixup in gpu and cpu versions

eginhard committed:
fix(vc): support both cpu and cuda (#244)

eginhard committed:
test: run tts trainings tests from python

eginhard committed:
refactor(vits): handle deprecation of multi-dimensional transpose

eginhard committed:
Merge pull request #208 from idiap/pytest

test: switch from nose2 to pytest

eginhard committed:
chore(api): improve type hints

eginhard committed:
feat(server): switch to python api

Server can then synthesise with any Coqui model, including XTTS, Fairseq,
Tortoise, Bark etc.

eginhard committed:
test(server): convert bash test to python

eginhard committed:
fix(bark): handle broken paths in config (#253)

eginhard committed:
build: update dev dependencies

eginhard committed:
build: drop python 3.9 support

eginhard committed:
refactor: enable ruff rule G004

eginhard committed:
refactor: apply safe automatic ruff lint fixes

Not manually checked. Generated with:
uv run ruff check tests/ TTS/ notebooks/ recipes/ --fix

eginhard committed:
refactor: apply unsafe automatic ruff lint fixes

Manually checked and adjusted after generating with:
uv run ruff check tests/ TTS/ notebooks/ recipes/ --fix --unsafe-fixes

eginhard committed:
style: run ruff format

eginhard committed:
build: switch black to ruff formatter

eginhard committed:
fix(configs): update config field types

eginhard committed:
build: update coqpit-config to 0.2.0

eginhard committed:
feat(vc): allow multiple target audio files

eginhard committed:
Merge pull request #221 from idiap/bokeh

Update plot_embeddings_umap notebook

eginhard committed:
Merge pull request #252 from idiap/server

Support all Coqui TTS models in the server

eginhard committed:
fix(openvoice): correctly set utterance length (#260)

eginhard committed:
docs(vc): add page on voice conversion

[ci skip]

eginhard committed:
chore: bump version to 0.25.2

[ci skip]

eginhard committed:
Merge pull request #256 from idiap/knnvc

Add kNN-VC

eginhard committed:
fix(fairseq): handle change of model file name

eginhard committed:
chore: bump version to 0.25.3

eginhard committed:
Merge pull request #264 from idiap/fix-fairseq

fix(fairseq): handle change of model file name

eginhard committed:
fix(tortoise): do not set device, which is handled in BaseTrainerModel

[ci skip]

eginhard committed:
Merge pull request #255 from idiap/drop-py39

Drop Python 3.9 support

eginhard committed:
test: parametrize multi-speaker tests

eginhard committed:
refactor(encoder): simplify code and run test from python

junland committed:
docs(server): clean up README (#272)

* docs: clean up server README

* docs: formatting

---------

Co-authored-by: Enno Hermann <enno.hermann@idiap.ch>

eginhard committed:
test: move integration tests into separate folder

eginhard committed:
fix(vocoder): add missing rank argument

eginhard committed:
Merge pull request #276 from idiap/pytest

Switch remaining CLI tests to Python, separate integration tests

eginhard committed:
build: add support for python 3.13

eginhard committed:
fix(vc): update allowed types in config

eginhard committed:
fix: notify users when wrong coqpit package is installed (#294)

shavit committed:
feat(server): add speaker_wav parameter for voice cloning (#295)

* Added speaker_wav parameter to the server

* Check if speaker_id is not empty

* feat(server): add input field for speaker_wav

---------

Co-authored-by: Enno Hermann <enno.hermann@idiap.ch>

eginhard committed:
feat(api): support setting speed (#316)

XTTS accepts a `speed` parameter, but that was previously captured in the API
which prevented it from being passed through. Removing the non-functional
`speed` parameter from the API lets it pass via `kwargs` to models that support
it.

eginhard committed:
refactor(transformers): LogitsWarper -> LogitsProcessor

https://github.com/huggingface/transformers/pull/35677

eginhard committed:
build: update transformers version

eginhard committed:
Merge pull request #319 from idiap/transformers

Refactor for compatibility with transformers>=4.47

DrewThomasson committed:
feat: add new persian-tts-female-vits model

DrewThomasson committed:
ci: add paths-ignore in workflows

eginhard committed:
chore: bump version to 0.26.0

eginhard committed:
fix(api): if there are only 2 model urls, assume they are model and config

eginhard committed:
fix(xtts): overwrite model_args in GPTTrainerConfig (#270)

This makes it possible to use --continue_path with XTTS

eginhard committed:
Merge pull request #332 from DrewThomasson/Persian-patch-1

Added new persian-tts-female-vits model

eginhard committed:
Merge pull request #334 from DrewThomasson/Workflow-patch-1

Added paths-ignore in workflows

fabiocat93 committed:
build: require numpy>=2

eginhard committed:
fix(numpy_transforms): support librosa 0.11.0

eginhard committed:
fix: numpy 2 compatibility

eginhard committed:
test(zoo): skip models that don't support weights-only loading

eginhard committed:
Merge pull request #346 from fabiocat93/dev

Switch to Numpy>=2, Pytorch>=2.3

eginhard committed:
build: remove spacy dependency

Spacy will likely be very slow to support Python 3.13, so it's better to let
users install it themselves if they need it. It's only used for text splitting
in XTTS, which is already disable by default.

eginhard committed:
fix(xtts): update colab finetuning notebook

eginhard committed:
test(data): adjust value to work in CI

1e-5 fails in CI for Python>=3.11

eginhard committed:
Merge pull request #377 from idiap/colab

fix(xtts): update colab finetuning notebook

btseee committed:
fix(forward_tts): ensure tensor 'g' is on the same device as 'x' (#378)

Sleuth56 committed:
feat(server): support multi-speaker models in MaryTTS endpoint (#351)

* Add support for multi-speaker models with the MaryTTS endpoint and commandline fallback

* Fixed the naming. Still working on the voices endpoint

* fixups

---------

Co-authored-by: Enno Hermann <enno.hermann@idiap.ch>

eginhard committed:
docs(server): better document usage and available parameters

eginhard committed:
chore: bump version to 0.26.1

eginhard committed:
Merge pull request #383 from idiap/spacy

Remove Spacy dependency

eginhard committed:
fix: don't convert to int to avoid constant value in onnx exports (#390)

NB:
TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!

eginhard committed:
build: restrict transformers to <4.52

Changes in 4.52 lead to corrupted output for XTTS

eginhard committed:
fix(demos.xtts): install compatible package versions

eginhard committed:
chore: bump version to 0.26.2

eginhard committed:
fix(xtts): clear error message when reference audio is too short

eginhard committed:
Merge pull request #396 from idiap/fix-xtts

XTTS bug fixes

KishoOoOoOo committed:
docs(docker): add docker-compose config

Co-authored-by: Enno Hermann <Eginhard@users.noreply.github.com>

eginhard committed:
docs(bark): fix parameter names

eginhard committed:
Merge pull request #411 from KishoOoOoOo/patch-1

docs: add Docker Compose config

eulphean committed:
fix(xtts): update finetuning Colab to support Gradio 5 (#408)

* Fix: Use gradio's textbox component to setup the callback for log parsing every 1 second.

* Fix: use the latest version of gradio

* fixup

---------

Co-authored-by: Enno Hermann <enno.hermann@idiap.ch>

eginhard committed:
fix(xtts): adapt gpt code to transformers 4.50+

- adapt to HF changes that moved a lot of code from generate() into separate
helper methods, so we don't need to duplicate that code in stream_generator.py
anymore
- removes some obsolete code to fall back to official HF implementations that
have e.g. improved caching support
- remove branches like beam-search in stream_generator.py that are never used to
shorten that code

eginhard committed:
fix(tortoise): adapt gpt code to transformers 4.50+

MarwanMashra committed:
build: add back Numpy 1.26 support (#413)

* add back numpy1 compatibility

* cleaner solution that works with numpy>=1.26

* adjustments based on PR feedback

* Update TTS/__init__.py

---------

Co-authored-by: Marwan <marwan@jumbomana.com>
Co-authored-by: Enno Hermann <Eginhard@users.noreply.github.com>

eginhard committed:
fix(xtts): pass attention mask also for streaming tts

eginhard committed:
refactor(bark): remove version checks around flash attention

scaled_dot_product_attention is now available in all Pytorch versions we support

eginhard committed:
fix(bark): add more detailed logs for model downloads

eginhard committed:
docs(faq): add info about the fork and fine-tuning

eginhard committed:
refactor(phonemizer): replace mecab-python3 with fugashi for japanese

Much simpler API

eginhard committed:
build: exclude fugashi 1.5.0

It accidentally provides a mecab package as well that conflicts with
python-mecab-ko: https://github.com/polm/fugashi/pull/110

eginhard committed:
Merge pull request #417 from idiap/fugashi

refactor(phonemizer): replace mecab-python3 with fugashi for japanese

eginhard committed:
Revert "build: exclude fugashi 1.5.0"

This reverts commit 75cc42a9e051214ce7ac7e126b83bbeb33f83220.

Obsolete because 1.5.0 has been yanked

eginhard committed:
fix(bark): disable weights-only loading

It doesn't seem possible to reliably do this with Numpy>2 because of
np._core.multiarray.scalar

eginhard committed:
build: update coqui-tts-trainer to 0.3.0 (#423)

* build: test new trainer branch

* refactor: update save_checkpoint calls

teddybear082 committed:
feat(server): add OpenAI-compatible endpoint (#421)

* add openai-compatible endpoint

-Add openai-compatible v1/audio/speech endpoint to server.py

-Allow openai server to use built in speaker voices, cloning wavs or directory with wavs as the voice

-Allow language_id as server.py command line argument in alternative to request parameter

-add usage examples to server.md

---------

Co-authored-by: Enno Hermann <enno.hermann@idiap.ch>

eginhard committed:
Merge pull request #414 from idiap/tf452

fix: update XTTS/Tortoise GPT code for HF transformers 4.52+

eginhard committed:
fix(configs): add default padding character

Otherwise tokenizer embeddings can be initialized with incorrect dimensions

eginhard committed:
chore: address linter issues

eginhard committed:
chore: remove double-defined parameters in GlowTTSConfig

eginhard committed:
fix(config): check for correct config type

eginhard committed:
feat: create mixin class for voice cloning

eginhard committed:
feat(xtts): support voice cloning mixin

eginhard committed:
feat: api support for voice cloning

eginhard committed:
docs: add dedicated page on voice cloning

eginhard committed:
feat(utils): add slugify function

eginhard committed:
chore(managers): remove unused code

eginhard committed:
refactor(managers): add type hints

eginhard committed:
docs: add cross-links and general improvements

eginhard committed:
feat(knnvc): support voice cloning mixin

eginhard committed:
refactor: torch.is_tensor(x) -> isinstance(x, torch.Tensor)

As recommended by Pytorch for better type checking

eginhard committed:
feat(openvoice): support voice cloning mixin

eginhard committed:
test: add test for voice cloning with caching

eginhard committed:
feat(freevc): support voice cloning mixin

eginhard committed:
feat(bark): support voice cloning mixin

eginhard committed:
chore(bark): remove unused code

Original source seems to be:
https://github.com/TronMetatron/bark

It's used there to filter out noisy audio, but Coqui doesn't actually do that.

eginhard committed:
fix(bark): use tensors throughout instead of numpy

eginhard committed:
feat(tortoise): support voice cloning mixin

eginhard committed:
fix(tortoise): add missing argument to fix voice cloning

eginhard committed:
fix(api): warn in tts_with_vc when model directly support cloning

eginhard committed:
Merge pull request #425 from idiap/default-pad

fix(configs): add default padding character

eginhard committed:
build: lower minimum pytorch version back to 2.1 (#432)

Now that Numpy 1 support is back, Pytorch 2.1 should also work

mehulanshumali committed:
fix(server): handle empty speaker_id, improve error handling (#436)

* Fix KeyError when speaker_id is empty string in TTS server

- Handle empty speaker_id and language_id parameters by converting to None
- Add input validation for text parameter
- Improve error handling for TTS synthesis failures
- Fixes issue #433: Reference audio input causes server to fail

* run `make style`

---------

Co-authored-by: Mehul Anshumali <mehul.a@phonepe.com>
Co-authored-by: Enno Hermann <enno.hermann@idiap.ch>

eginhard committed:
fix: clone tensors that could be stored as inference tensors

Otherwise the tensors cannot be reused in training.

eginhard committed:
chore: remove unused code

eginhard committed:
refactor!: add synthesize() method to every TTS model

eginhard committed:
refactor: move redundant methods into parent class

eginhard committed:
fix(vocoder): match scheduler with correct optimizer

eginhard committed:
feat: cloned voice caching support for d-vector models

eginhard committed:
Merge pull request #438 from idiap/cloning

Speaker caching for cloned voices

eginhard committed:
feat: indicate in model config whether cloning is supported

eginhard committed:
feat: make use of supports_cloning attribute

eginhard committed:
fix(xtts): correctly parse kwargs

eginhard committed:
chore: bump version to 0.27.0

eginhard committed:
Merge pull request #453 from idiap/refactor-synthesis

feat!: unify synthesize() methods

eginhard committed:
ci(docker): update deprecated commands, update actions

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

ivuorio committed:
fix: register custom formatters just like built-in ones (#293)

* add new methos for handling custom formatters

* add docstring

* add unittest for the custom formatters

* refactor(datasets): register formatters in dict

---------

Co-authored-by: Enno Hermann <enno.hermann@idiap.ch>

eginhard committed:
fix(bark): allow loading npz voices from the original model

eginhard committed:
fix(recipes): wrap scripts with main() functions (#462)

eginhard committed:
fix(bark): ensure all tensors are on the same device

eginhard committed:
fix(tortoise): remove low vram mode (#464)

It wasn't really faster than CPU-only and caused issues with high vram mode.

eginhard committed:
Merge pull request #461 from idiap/bark-npz

fix(bark): allow loading npz voices from the original model

eginhard committed:
build: restrict to torch/torchaudio<2.9

torchaudio 2.9 removes crucial functionality:
https://github.com/pytorch/audio/issues/3902

eginhard committed:
chore: bump version to 0.27.1

eginhard committed:
Merge pull request #488 from idiap/torch29

build: restrict to torch/torchaudio<2.9

eginhard committed:
test(text): make espeak-ng default backend

eginhard committed:
ci(docker): update ubuntu/cuda/python versions

eginhard committed:
fix(utils.manage): use context manager for tqdm

eginhard committed:
ci(docker): switch to uv

zkhan122 committed:
display offending text going over limit

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/coqui-ai/TTS?pullRequest=4336) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/coqui-ai/TTS?pullRequest=4336) before we can accept your contribution.<br/>**1** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: Olexandr88<br/>:x: ivuorio<br/>:x: zkhan122<br/>:x: langfod<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/coqui-ai/TTS?pullRequest=4336) it.</sub>

eginhard commented:
Thanks for the PR! This repository is not maintained anymore, but feel free to open a PR in our fork: https://github.com/idiap/coqui-ai-TTS Just make sure to use logging instead of a `print()`.

zkhan122 committed:
display offending text going over limit

zkhan122 committed:
display offending text going over limit

zkhan122 commented:
> Thanks for the PR! This repository is not maintained anymore, but feel free to open a PR in our fork: https://github.com/idiap/coqui-ai-TTS Just make sure to use logging instead of a `print()`.

Hi, I have now done so

Olexandr88 committed:
docs(README): fix Double Decoder Consistency link which was not working (#500)

* Update README.md

* Update README.md

eginhard committed:
build: restrict transformers to <4.56

Streaming XTTS is broken again

eginhard committed:
chore: bump version to 0.27.2

eginhard committed:
build: add explicit dependency on numba

Ensures that a sufficiently old version is picked in CI when running with
--resolution lowest-direct to avoid compatibility issues.

eginhard committed:
Merge pull request #285 from idiap/py313

Add support for Python 3.13

langfod committed:
refactor(phonemizer): replace jieba with spacy_pkuseg

eginhard committed:
test(phonemizer): add test cases for chinese

eginhard committed:
Merge pull request #504 from langfod/spacy-pguseg

Replace jieba with spacy_pkuseg

eginhard committed:
Update TTS/tts/layers/xtts/tokenizer.py

eginhard committed:
Update TTS/tts/layers/xtts/tokenizer.py

eginhard committed:
Update TTS/tts/layers/xtts/tokenizer.py

eginhard committed:
Merge branch 'dev' into add-over-limit-text-display
